src/components/Searchbar/SearchBar.tsx

```diff
@@ -1,94 +1,95 @@
-import { useState } from "react";
+// import { useState } from "react";
 import TextField from "@mui/material/TextField";
 import Button from "@mui/material/Button";
 import styles from "../Searchbar/Searchbar.module.css";
-import Select, { SelectChangeEvent } from "@mui/material/Select";
+import Select from "@mui/material/Select";
 import MenuItem from "@mui/material/MenuItem";
 import FormControl from "@mui/material/FormControl";
 import InputLabel from "@mui/material/InputLabel";
-
-// import InputLabel from "@mui/material/InputLabel";
+import { useForm, SubmitHandler } from "react-hook-form";
 
 interface Props {
-  getSearchQueryName: (searchQueryName: string) => void;
-  getSearchQueryYear: (searchQueryYear: string) => void;
-  getSearchQueryType: (searchQueryType: string) => void;
+  getSearchQueryName: (title: string) => void;
+  getSearchQueryYear: (year: string) => void;
+  getSearchQueryType: (type: string) => void;
 }
+
+type Inputs = {
+  title: string;
+  year: string;
+  type: string;
+};
+
 export function Searchbar({
   getSearchQueryName,
   getSearchQueryYear,
   getSearchQueryType,
 }: Props) {
-  const [searchQueryName, setSearchQueryName] = useState("");
-  const [searchQueryYear, setSearchQueryYear] = useState("");
-  const [searchQueryType, setSearchQueryType] = useState("");
+  const {
+    register,
+    handleSubmit,
+    reset,
 
-  const handleOnSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-
-    getSearchQueryName(searchQueryName);
-    getSearchQueryYear(searchQueryYear);
-    getSearchQueryType(searchQueryType);
-    setSearchQueryName("");
-    setSearchQueryYear("");
-    setSearchQueryType("");
-  };
+    formState: { errors },
+  } = useForm<Inputs>({ mode: "onBlur" });
 
-  const handleOnChange = (
-    e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
-  ) => {
-    switch (e.target.name) {
-      case "title":
-        setSearchQueryName(e.target.value);
-        break;
-      case "year":
-        setSearchQueryYear(e.target.value);
-        break;
-      default:
-        return;
-    }
-  };
-
-  const handleChange = (event: SelectChangeEvent) => {
-    setSearchQueryType(event.target.value as string);
+  const onSubmit: SubmitHandler<Inputs> = (data) => {
```

Comment:
When using react-hook-form, you must remove the local state with search queries. Because form does it instead of you. So you have to refactor this part. Remove the local state, and instead of useEffect in MovieGallery create a handleSubmit function that will receive data from the form and pass it to the function fetchMovies. So you also don't need duplicate search queries into your apiService they will receive via props.

Reply:
const handleSubmit = (data) => {
    apiService.fetchMovies(data).then ....
}